### PR TITLE
Ria 3951 remove legal rep

### PR DESCRIPTION
**Pull request by bradsorour**
### JIRA link (if applicable) ###

[RIA-3951](https://tools.hmcts.net/jira/browse/RIA-3951)

### Change description ###

See Jira

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RemoveRepresentationConfirmationTest.java [hunk 1]
```diff
@@ -52,18 +51,8 @@ void should_apply_noc_and_return_confirmation() {
             removeRepresentationConfirmation.handle(callback);
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RemoveRepresentationConfirmationTest.java:51]: Test should include for event REMOVE_LEGAL_REPRESENTATIVE
Suggest using @ParameterizedTest for event

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RemoveRepresentationPreparerTest.java [hunk 2]
```diff
@@ -128,7 +128,7 @@ void it_can_handle_callback() {
 
                 boolean canHandle = removeRepresentationPreparer.canHandle(callbackStage, callback);
 
-                if (event == Event.REMOVE_REPRESENTATION
+                if ((event == Event.REMOVE_REPRESENTATION || event == Event.REMOVE_LEGAL_REPRESENTATIVE)
                     && callbackStage == PreSubmitCallbackStage.ABOUT_TO_START) {
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RemoveRepresentationPreparerTest.java:132]: Test should_write_to_change_organisation_request_field()
 handling_should_throw_if_cannot_actually_handle() 
 should_respond_with_error_when_organisation_policy_not_present()
should include for event REMOVE_LEGAL_REPRESENTATIVE
Suggest using @ParameterizedTest for event

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/SendPostNotificationHandlerTest.java [hunk 3]
```diff
@@ -78,7 +78,8 @@ void it_can_handle_callback() {
```
Comment (Lee365) [inline src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/SendPostNotificationHandlerTest.java:78]: Test should_notify_case_officer_that_case_is_listed()
should include for event REMOVE_LEGAL_REPRESENTATIVE, REMOVE_REPRESENTATION
Suggest using @ParameterizedTest for event

**Review by Lee365 — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RemoveRepresentationConfirmationTest.java (hunk 1 above)
Comment (bradsorour) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/RemoveRepresentationConfirmationTest.java:50]: Done

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/SendPostNotificationHandlerTest.java (hunk 3 above)
Comment (bradsorour) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/postsubmit/SendPostNotificationHandlerTest.java:78]: Good spot, included for the other events

File: src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RemoveRepresentationPreparerTest.java (hunk 2 above)
Comment (bradsorour) [reply src/test/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RemoveRepresentationPreparerTest.java:139]: Done for the relevant tests

**Review by Lee365 — DISMISSED** [review]
LGTM

**Review by Srihari-Nakkala — DISMISSED** [review]
LGTM

**Review by makowalski — APPROVED** [review]
LGTM

**Pull request merged**